cumulusci/core/keychain/BaseEncryptedProjectKeychain.py

```diff
@@ -71,7 +71,19 @@ def _decrypt_config(self, config_class, encrypted_config, extra=None):
         iv = encrypted_config[:16]
         cipher, iv = self._get_cipher(iv)
         pickled = cipher.decrypt(encrypted_config[16:])
-        config_dict = pickle.loads(pickled)
+        try:
+            unpickled = pickle.loads(pickled, encoding="bytes")
+        except TypeError:  # Python 2
+            config_dict = pickle.loads(pickled)
+        else:  # Python 3
+            # Convert pickles created in Python 2
+            config_dict = {}
+            for k, v in unpickled.items():
+                if isinstance(k, bytes):
+                    k = k.decode("utf-8")
+                if isinstance(v, bytes):
+                    v = v.decode("utf-8")
+                config_dict[k] = v
```

Comment:
Note to self: there might be problems going the other direction too if you're trying to switch back and forth between Python 2 and Python 3 on the same machine.

Reply:
@cdcarter I changed `_encrypt_config` to use pickle protocol 2 which Python 3 can read, and added similar code to translate unicode back to bytes when going in that direction. We may uncover more issues here but it worked going back and forth with a scratch org for me.

Reply:
@davisagli - I'll give it a test on my very weird-organically-grown-many-years keychain on my main machine on Monday, see what I can and cannot roundtrip...

I'm struggling to find muuuch documentation on pickle protocol 3 as it wasn't introduced with a PEP but just as a part of py3k... I just want to sanity check that we aren't creating a footgun for ourselves in the future by continuing to serialize in protocol 2. Based on my understanding, we should be able to switch to protocol 3 without issue once py3 is a minimum requirement. Is there any reason to put the pickle protocol version into [env var|module global export|global yaml] for configuration or runtime inspection?

Reply:
My understanding is that the main change in protocol 3 was explicit support for serializing bytes as a distinct type from strings. I'm not too worried about using protocol 2; I wouldn't expect Python to remove support for it any time soon -- and yeah, I see this as a temporary measure until we update CumulusCI to only support Python 3.

As written this code is imperfect; it only translates top-level keys/values. Perhaps it should recurse.